File: nature-harvest-frontend/src/app/responses/cart/cart.response.ts
import { ProductResponse } from '../product/product.response';

export class CartResponse {
  id: number;
  userId: string;
  product: ProductResponse;
  quantity: number;

  constructor(data: any) {
    this.id = data.id;
    this.userId = data.userId;
    this.product = data.product;
    this.quantity = data.quantity;
  }
}

File: nature-harvest-backend/nature-harvest-backend/src/main/java/com/api/nature_harvest_backend/responses/cart/CartResponse.java
package com.api.nature_harvest_backend.responses.cart;

import com.api.nature_harvest_backend.models.Cart;
import com.api.nature_harvest_backend.models.Product;
import com.api.nature_harvest_backend.responses.product.ProductResponse;
import lombok.*;

import java.util.List;
import java.util.stream.Collectors;


@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class CartResponse {
    private Long id;
    private String userId;
    private ProductResponse product;
    private int quantity;

    public static List<CartResponse> fromCart(List<Cart> cartList) {
        return cartList.stream().map(cart -> {
            return CartResponse.builder()
                  .id(cart.getId())
                  .userId(cart.getUser().getId())
                  .product(ProductResponse.fromProduct(cart.getProduct()))
                  .quantity(cart.getQuantity())
                  .build();
        }).collect(Collectors.toList());
    }
}
